### PR TITLE
Edge cases and Tests

### PR DESCRIPTION
**Pull request by keronei**
This is an implementation of various edge cases in order to keep the api's at check and prevent crashes during runtime.
The following is an highlight of what is taken care off:

1. User attempting to create an entry while providing less params.
2. Requesting deletion without providing an `identifier`(key) for the particular entry.
3. Attempting update to an entry without the `identifier`. 
4. Differentiating `GET` requests for a single entry and all entries, this employs an identifier param that is set to `null`, which by default serves the user with all entries, however, when a the param is supplied, the user is replied with a single entry, in cases where a match is successfully found.

This PR furthermore contains tests for the particular cases that have been handled.

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -1,7 +1,14 @@
+
 from flask import Flask, Blueprint
 from .api.v1 import version_one as v1
+from .api.v2 import version_two as v2
+from .config import app_config
+from .api.v2.models import db
 
 def launcher():
     app = Flask(__name__)
+    app.config.from_object(app_config['development'])
```
Comment (thoth-ky) [inline app/__init__.py]: make this dynamic. As it is it will always open the app in `development` environment

File: app/tests/v1/test.py [hunk 2]
```diff
@@ -6,69 +6,92 @@
 
 app = launcher()
```
Comment (thoth-ky) [inline app/tests/v1/test.py:5]: At this point you should be passing the environment, in this case `testing` so that it picks up relevant config. In line with changes suggested in the launcher function

File: app/tests/v1/test.py [hunk 3]
```diff
@@ -6,69 +6,92 @@
...
+                                "comment":"Requires imediate attention",
+                                "location":"90,0",
+                                "status":"Pending"
+                                
+                             }
+                            
+    def test_create_new_flag(self):
+        with app.test_request_context():
+            response=self.app.post('/api/v1/red-flag',headers={'Content-Type': 'application/json'},data=json.dumps(self.data) )
+            self.assertEqual(response.status_code,200,"Success on post should be 204")
```
Comment (thoth-ky) [inline app/tests/v1/test.py]: Test contents of the response. Not just the status code

File: app/tests/v1/test.py [hunk 4]
```diff
@@ -6,69 +6,92 @@
...
     def test_fetching_all_flags(self):
-        self.app = app.test_client()
-        response = self.app.get("/api/v1/red-flag")
-        self.assertEqual(response.status_code, 200, "Sucessfull query returns 200")
-        
+        with app.test_request_context():
```
Comment (thoth-ky) [inline app/tests/v1/test.py:34]: NOt necessary if you make your `launcher` function dynamic

File: app/tests/v1/test.py [hunk 5]
```diff
@@ -6,69 +6,92 @@
...
+            self.assertEqual(response.status_code, 200, "Sucessfull query returns 200")
+            
     def test_get_single_flag(self):
-        self.app = app.test_client()
-        response = self.app.get("/api/v1/red-flag/23243")
-        self.assertEqual(response.status_code, 200, "Status has to be 200")
+        with app.test_request_context():
+            response = self.app.get("/api/v1/red-flag/23243")
+            self.assertEqual(response.status_code, 200, "Status has to be 200")
     def test_deletion(self):
```
Comment (thoth-ky) [inline app/tests/v1/test.py]: deletion of what? Make your test names self documenting like `test_can_delete_red_flag`

**Review by thoth-ky — CHANGES_REQUESTED** [review]
Kindly act on the feedback I have left below.

**Comment by thoth-ky** [comment]
Add test for more edge cases, like what happens when one tries to fetch some record that is not there or gives unvalidated data to create a record?

File: app/__init__.py (hunk 1 above)
Comment (keronei) [reply app/__init__.py]: Have implemented in the this format:

`app = launcher(app_config["development"])`

Which will take Production environment if no specification is made.

File: app/tests/v1/test.py (hunk 3 above)
Comment (keronei) [reply app/tests/v1/test.py]: Have included assertions for the expected content.

File: app/tests/v1/test.py (hunk 4 above)
Comment (keronei) [reply app/tests/v1/test.py:34]: Haven't figured out how to have the function dynamic, but will seek advise of the same and implement.

File: app/tests/v1/test.py (hunk 5 above)
Comment (keronei) [reply app/tests/v1/test.py]: Have refactored the namings to speak for themselves. With meaningful names now.

**Review by thoth-ky — APPROVED** [review]
Add travis on this branch so that we kow if the tests are passing.

File: app/__init__.py [hunk 6]
```diff
@@ -1,7 +1,16 @@
+
 from flask import Flask, Blueprint
 from .api.v1 import version_one as v1
+from .api.v2 import version_two as v2
```
Comment (thoth-ky) [inline app/__init__.py]: without a v2, this will throw an error

File: requirements.txt [hunk 7]
```diff
@@ -1,21 +1,29 @@
...
 pluggy==0.8.0
+psycopg2==2.7.6.1
 py==1.7.0
 pytest==4.0.1
 pytest-cov==2.6.0
+python-dateutil==2.7.5
+python-editor==1.0.3
 pytz==2018.7
 six==1.11.0
+SQLAlchemy==1.2.14
```
Comment (thoth-ky) [inline requirements.txt]: Remove this because we will not be using ORM

File: requirements.txt [hunk 8]
```diff
@@ -1,21 +1,29 @@
...
 aniso8601==4.0.1
 atomicwrites==1.2.1
 attrs==18.2.0
 Click==7.0
 coverage==4.5.2
 Flask==1.0.2
 Flask-API==1.0
+Flask-Migrate==2.3.1
 Flask-RESTful==0.3.6
+Flask-SQLAlchemy==2.3.2
```
Comment (thoth-ky) [inline requirements.txt]: This too

File: requirements.txt [hunk 9]
```diff
@@ -1,21 +1,29 @@
+alembic==1.0.5
```
Comment (thoth-ky) [inline requirements.txt]:
```suggestion
```
Remove all ORM related dependencies

**Review by thoth-ky — CHANGES_REQUESTED** [review]
Act on feedback left. Also mantainability is pretty low
